docs/todo-due-dates.md

```diff
@@ -0,0 +1,181 @@
+- Feature Name: `todo-due-dates`
+- Start Date: `2020-11-09`
+- RFC PR: (leave this empty)
+- Checkup Issue: (leave this empty)
+
+# Summary
+
+[summary]: #summary
+
+This RFC proposes enhancing the todo functionality with due days, allowing authors to add accountability and commitment to fix the todo items by a specified amount of time.
+
+# Motivation
+
+[motivation]: #motivation
+
+The lint todo functionality is intended to seamlessly integrate with existing linting tools, allowing the incremental fixing of linting errors on very large projects. However, one undesirable side effect of this is that developers may forget about fixing the todo items, leading to the accumulation of technical debt.
+
+By adding a day threshold to the todo items, developers will now have a commitment to fix the linting problems because todo items that are past this threshold will have their severity increased accordingly.
+
+# How We Will Teach This
+
+[pedagogy]: #pedagogy
+
+- There will be documentation on how to configure when todo items are due to be fixed.
+- There will be a config and/or CLI argument to specify the number of days an item's severity changes, either to `warn` or to `error`.
+
+# Requirements
+
+[requirements]: #requirements
+
+- When no due date configuration is present (primarily configured in package.json), todos are added with no due dates.
+- Authors can specify the number of days the todos become errors after its creation date.
+- Author can also specify the number of days the todos become warnings after its creation date. The error threshold must also be passed and must be greater than this threshold.
+- Configuration can be specified in a common file or overriden via CLI arguments.
+
+# Implementation
+
+[requirements]: #implementation
+
+## High Level Flow
+
+When an author runs a linter with the todo functionality, all errors are converted to todos with no due date threshold unless one is present in `package.json` _or_ passed as a command line argument.
+
+- `daysToWarn` - number of days after its creation date that a todo transitions into a `warn`
+- `daysToError` number of days after its creation date that a todo transitions into an `error`.
+- `daysToWarn` _and_ `daysToError` - if used, `daysToError` must be greater than `daysToWarn`.
+
+## Workflows
+
+### Normal todo workflow
+
+- No due date is expected when adding todo items with no threshold provided in the `package.json` or the CLI arguments `daysToError` or `daysToWarn`.
+
+### Todos that are past-due become errors
+
+Example use case: 20 days for todos to be considered errors since its creation date.
+
+**Via config:**
+
+In `package.json`:
+
+```json
+{
+  "lintTodo": {
+    "daysToDecay": {
+      "warn": 30,
+      "error": 90
+    }
+  }
+}
+```
```

Comment:
Are these warn/error numbers correct?

Reply:
In what sense? They're examples, so I'm not sure what correctness you're referring to.

Reply:
These are just examples, there will not be any default value (from the OSS side) but if folks want to make their own linting setup that sets defaults this is how they'd do it.